services/vmalert/vmalert.go

```diff
@@ -76,33 +76,33 @@ func NewVMAlert(externalRules *ExternalRules, baseURL string) (*Service, error)
 		client: &http.Client{
 			Transport: t,
 		},
-		l:    logrus.WithField("component", "vmalert"),
-		irtm: irtm,
-		sema: make(chan struct{}, 1),
+		l:        logrus.WithField("component", "vmalert"),
+		irtm:     irtm,
+		reloadCh: make(chan struct{}, 1),
 	}, nil
 }
 
-// Describe implements prometheus.Collector.
-func (svc *Service) Describe(ch chan<- *prom.Desc) {
-	svc.irtm.Describe(ch)
-}
-
-// Collect implements prometheus.Collector.
-func (svc *Service) Collect(ch chan<- prom.Metric) {
-	svc.irtm.Collect(ch)
-}
-
 // Run runs VMAlert configuration update loop until ctx is canceled.
 func (svc *Service) Run(ctx context.Context) {
+	// If you change this and related methods,
+	// please do similar changes in alertmanager and victoriametrics packages.
+
 	svc.l.Info("Starting...")
 	defer svc.l.Info("Done.")
 
+	// reloadCh, configuration update loop, and RequestConfigurationUpdate method ensure that configuration
+	// is reloaded when requested, but several requests are batched together to avoid too often reloads.
+	// That allows the caller to just call RequestConfigurationUpdate when it seems fit.
+	if cap(svc.reloadCh) != 1 {
```

Comment:
this channel is always buffered then why do we have this check?

Reply:
To prevent someone from changing buffer size without reading that comment